stdlib/public/runtime/Metadata.cpp

```diff
@@ -3484,6 +3484,10 @@ ExistentialMetatypeValueWitnessTableCacheEntry(unsigned numWitnessTables) {
 SWIFT_RUNTIME_EXPORT
 const ExistentialMetatypeMetadata *
 swift::swift_getExistentialMetatypeMetadata(const Metadata *instanceMetadata) {
+  if (instanceMetadata->getKind() != MetadataKind::Existential
+      && instanceMetadata->getKind() != MetadataKind::ExistentialMetatype)
+    return nullptr;
```

Comment:
All of the other (non-metatype-decoding) callers to `swift_getExistentialMetatypeMetadata` already ensure that they provide correct instance metadata, so this check is redundant and the result can be assumed to never be NULL. I think this check should be lifted up into the caller(s) in the demangle-to-metadata path.

Reply:
Done.